avm/res/network/trafficmanagerprofile/main.bicep

```
@@ -0,0 +1,246 @@
+metadata name = 'Traffic Manager Profiles'
+metadata description = 'This module deploys a Traffic Manager Profile.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the Traffic Manager.')
+@minLength(1)
+param name string
+
+@description('Optional. The status of the Traffic Manager profile.')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param profileStatus string = 'Enabled'
+
+@description('Optional. The traffic routing method of the Traffic Manager profile.')
+@allowed([
+  'Performance'
+  'Priority'
+  'Weighted'
+  'Geographic'
+  'MultiValue'
+  'Subnet'
+])
+param trafficRoutingMethod string = 'Performance'
+
+@description('Required. The relative DNS name provided by this Traffic Manager profile. This value is combined with the DNS domain name used by Azure Traffic Manager to form the fully-qualified domain name (FQDN) of the profile.')
+param relativeName string
```

Could we make this nullable and then coalesce to name in the resource deployment logic? 

```suggestion
@description('Optional. The relative DNS name provided by this Traffic Manager profile. This value is combined with the DNS domain name used by Azure Traffic Manager to form the fully-qualified domain name (FQDN) of the profile.')
param relativeName string?
```